src/components/DashboardLayout/NavItem.jsx

```diff
@@ -102,9 +104,26 @@ const NavItem = (props) => {
     hoverMenuItem = false,
     defaultExpanded = false,
   } = props;
-  
+
   const [expanded, setExpanded] = useState(defaultExpanded);
 
+  // Check if the text overflows
+  useEffect(() => {
```

Comment:
We don't need to add a window resize listener as the nav item size is fixed an independent of window size